sqlair.go

```diff
@@ -163,3 +164,61 @@ func (q *Query) One(outputArgs ...any) error {
 	iter.Decode(outputArgs...)
 	return iter.Close()
 }
+
+// All iterates over the query and decodes all the rows.
+// resultColumns must contain pointers to slices of each of the query arguments e.g.
+// > q.All(&[]*Person{}, &[]*Address{}) or q.All(&[]Address{}, &[]Person{})
```

Comment:
This is slightly confusing because these values are being immediately discarded, which is something we do for real in the case of Prepare, but here it would make no sense as the whole point is retrieving those results. Maybe:
```
// All iterates over the query and decodes all rows into the provided slices.
//
// For example:
//
//     var pslice []Person
//     var aslice []*Address{}
//     err := query.All(&pslice, &aslice)
// 
func ...
```

We should really take []Person here rather than []*Person, otherwise we'll be allocating a slice value that points to individually allocated Person values, instead of having one slice of contiguous values which is significantly more efficient.

Reply:
Thats a good point, should we allow both `[]*Person` and `[]Person` or just `Person[]`. I feel like it might be best to give the user the option, but steer them towards `[]Person` in the example.

Reply:
I'll assume from the latter comments that we want to allow both.